sw/device/tests/flash_ctrl_idle_low_power_test.c

```diff
@@ -0,0 +1,117 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_flash_ctrl.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/flash_ctrl_testutils.h"
+#include "sw/device/lib/testing/pwrmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+enum {
+  kFlashDataRegion = 0,
+  kRegionBasePageIndex = 256,  // First page in bank 1 (avoids program code.)
+  kPartitionId = 0,
+  kRegionSize = 1,
+  kNumWords = 128,
+};
+
+bool test_main(void) {
+  dif_flash_ctrl_state_t flash;
+  dif_pwrmgr_t pwrmgr;
+  dif_rstmgr_t rstmgr;
+  dif_aon_timer_t aon;
+
+  CHECK_DIF_OK(dif_flash_ctrl_init_state(
+      &flash, mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)));
+  CHECK_DIF_OK(dif_pwrmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_PWRMGR_AON_BASE_ADDR), &pwrmgr));
+  CHECK_DIF_OK(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_EARLGREY_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  CHECK_DIF_OK(dif_aon_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR), &aon));
+
+  CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon));
+  CHECK_DIF_OK(dif_pwrmgr_set_request_sources(&pwrmgr, kDifPwrmgrReqTypeReset,
+                                              kDifPwrmgrResetRequestSourceTwo,
+                                              kDifToggleEnabled));
+
+  dif_rstmgr_reset_info_bitfield_t rstmgr_reset_info;
+  CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rstmgr_reset_info));
+  CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+
+  uint32_t address = flash_ctrl_testutils_data_region_setup(
+      &flash, kRegionBasePageIndex, kFlashDataRegion, kRegionSize);
+
+  uint32_t readback_data[kNumWords];
+
+  if (rstmgr_reset_info == kDifRstmgrResetInfoPor) {
+    // Create data. Simple count for data is sufficient which will
+    // be different than the 0xFFFFFFFF that is created
+    // with an erase.
+    uint32_t data[kNumWords];
+    for (int i = 0; i < kNumWords; ++i) {
+      data[i] = i;
```

Comment:
Randomize the data using `rand_testutils_gen32()`:
https://github.com/lowRISC/opentitan/blob/09ef94e1a7a3f8d99854537f2e201b219e39fefc/sw/device/lib/testing/rand_testutils.c#L18

Reply:
Change made, thanks.